src/caret_analyze/architecture/struct/node.py

```diff
@@ -248,6 +248,28 @@ def assign_variable_passings(self, callback_name_write: str, callback_name_read:
                     for passing in self._variable_passings_info]:
             self._variable_passings_info.append(passing)
 
+    # def remove_message_context(self, node_name: str, context_type: str,
+    #                            sub_topic_name: str, pub_topic_name: str):
+    #     # To assign message context, update_node_path() is called in Architecture.
+    #     # This is because module dependency.
+    #     # TODO: Refactoring module dependency
+    #     pass
+
+    def remove_publisher_and_callback(self, publish_topic_name: str, callback_name: str):
+        callback: CallbackStruct = \
+            Util.find_one(lambda x: x.callback_name == callback_name, self.callbacks)
+        callback.remove_publisher(publish_topic_name)
+
+        publisher: PublisherStruct = \
+            Util.find_one(lambda x: x.topic_name == publish_topic_name, self._publishers)
+        publisher.remove_callback(callback)
+
+    def remove_variable_passing(self, callback_name_write: str, callback_name_read: str):
+        self._variable_passings_info =\
+            [passing for passing in self._variable_passings_info or []
+             if (callback_name_write, callback_name_read) !=
+             (passing.callback_name_write, passing.callback_name_read)]
```

Comment:
The following is more clearer.

```suggestion
    def remove_variable_passing(self, callback_name_write: str, callback_name_read: str):
        if self._variable_passings_info:
            self._variable_passings_info =\
                [passing for passing in self._variable_passings_info
                 if (callback_name_write, callback_name_read) !=
                 (passing.callback_name_write, passing.callback_name_read)]
```

Reply:
Fix it in [54ed564](https://github.com/tier4/CARET_analyze/pull/297/commits/54ed56419cac94aaedc2bf635f24f10927e017b8).